### PR TITLE
Fix #1261 - remove unused libc function checkers

### LINKED ISSUE #1261: Migrate function checks from headers to Meson
Currently we have code like
```c
#ifndef HAVE_EPRINTF
#define eprintf(...) fprintf(stderr, __VA_ARGS__)
#define HAVE_EPRINTF 1
#endif
```
in `librz/include/rz_types.h`.
It should be migrated to the `meson.build` checks like it's done for some functions already:
```meson
foreach item : [
    ['arc4random_uniform', '#include <stdlib.h>', []],
    ['explicit_bzero', '#include <string.h>', []],
    ['explicit_memset', '#include <string.h>', []],
    ['clock_nanosleep', '#include <time.h>', []],
    ['clock_gettime', '#include <time.h>', [lrt]],
    ['sigaction', '#include <signal.h>', []],
    ['pipe', '#include <unistd.h>', []],
    ['execv', '#include <unistd.h>', []],
    ['execve', '#include <unistd.h>', []],
    ['execvp', '#include <unistd.h>', []],
    ['execl', '#include <unistd.h>', []],
    ['system', '#include <stdlib.h>', []],
    ['fork', '#include <unistd.h>', []],
    ['pipe2', '#define _GNU_SOURCE\n#include <fcntl.h>\n#include <unistd.h>', []]
  ]
  func = item[0]
  ok = cc.has_function(func, prefix: item[1], dependencies: item[2])
  userconf.set10('HAVE_@0@'.format(func.to_upper()), ok)
endforeach
```

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Migrated C preprocessor checks to the meson-provided checks.
- Removed unused checks.

**Test plan**

CI is green

Please be sure to check also locally. cc @devnexen @GustavoLCR @thestr4ng3r 

**Closing issues**

Closes https://github.com/rizinorg/rizin/issues/1261

### CODE REVIEW

**Comment by devnexen** [comment]
verified on openbsd

**Review by wargio — APPROVED** [review]

**Comment by XVilka** [comment]
Not sure why all of a sudden there is a tree-sitter runtime error on Debian Wheezy:
```
 ../subprojects/tree-sitter-c/src/parser.c:4147:3: warning: variable 'eof' set but not used [-Wunused-but-set-variable]
[936/1667] Linking static target subprojects/tree-sitter-c/libtree-sitter-c.a
[937/1667] Compiling C object librz/type/librz_type.so.0.3.0-git.p/format.c.o
[938/1667] Linking target librz/type/librz_type.so.0.3.0-git
FAILED: librz/type/librz_type.so.0.3.0-git 
cc  -o librz/type/librz_type.so.0.3.0-git librz/type/librz_type.so.0.3.0-git.p/base.c.o librz/type/librz_type.so.0.3.0-git.p/format.c.o librz/type/librz_type.so.0.3.0-git.p/function.c.o librz/type/librz_type.so.0.3.0-git.p/helpers.c.o librz/type/librz_type.so.0.3.0-git.p/path.c.o librz/type/librz_type.so.0.3.0-git.p/serialize_functions.c.o librz/type/librz_type.so.0.3.0-git.p/serialize_types.c.o librz/type/librz_type.so.0.3.0-git.p/type.c.o librz/type/librz_type.so.0.3.0-git.p/typeclass.c.o librz/type/librz_type.so.0.3.0-git.p/parser_c_cpp_parser.c.o librz/type/librz_type.so.0.3.0-git.p/parser_types_parser.c.o librz/type/librz_type.so.0.3.0-git.p/parser_types_storage.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,librz_type.so.0.3.0-git '-Wl,-rpath,$ORIGIN/../util' -Wl,-rpath-link,/__w/rizin/rizin/rizin/build/librz/util librz/util/librz_util.so.0.3.0-git -Wl,--start-group subprojects/tree-sitter-0.19.4/libtree_sitter.a subprojects/tree-sitter-c/libtree-sitter-c.a -Wl,--end-group
subprojects/tree-sitter-0.19.4/libtree_sitter.a(lib_src_lib.c.o): In function `clock_now':
/__w/rizin/rizin/rizin/build/../subprojects/tree-sitter-0.19.4/lib/src/././clock.h:73: undefined reference to `clock_gettime'
/__w/rizin/rizin/rizin/build/../subprojects/tree-sitter-0.19.4/lib/src/././clock.h:73: undefined reference to `clock_gettime'
collect2: error: ld returned 1 exit status
[939/1667] Compiling C object librz/analysis/librz_analysis.so.0.3.0-git.p/cond.c.o
```

**Comment by XVilka** [comment]
@ret2libc failures are unrelated:
- https://github.com/rizinorg/rizin/issues/1344

```
c  -o librz/type/librz_type.so.0.3.0-git librz/type/librz_type.so.0.3.0-git.p/base.c.o librz/type/librz_type.so.0.3.0-git.p/format.c.o librz/type/librz_type.so.0.3.0-git.p/function.c.o librz/type/librz_type.so.0.3.0-git.p/helpers.c.o librz/type/librz_type.so.0.3.0-git.p/path.c.o librz/type/librz_type.so.0.3.0-git.p/serialize_functions.c.o librz/type/librz_type.so.0.3.0-git.p/serialize_types.c.o librz/type/librz_type.so.0.3.0-git.p/type.c.o librz/type/librz_type.so.0.3.0-git.p/typeclass.c.o librz/type/librz_type.so.0.3.0-git.p/parser_c_cpp_parser.c.o librz/type/librz_type.so.0.3.0-git.p/parser_types_parser.c.o librz/type/librz_type.so.0.3.0-git.p/parser_types_storage.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,librz_type.so.0.3.0-git '-Wl,-rpath,$ORIGIN/../util' -Wl,-rpath-link,/__w/rizin/rizin/rizin/build/librz/util librz/util/librz_util.so.0.3.0-git -Wl,--start-group subprojects/tree-sitter-0.19.4/libtree_sitter.a subprojects/tree-sitter-c/libtree-sitter-c.a -Wl,--end-group
subprojects/tree-sitter-0.19.4/libtree_sitter.a(lib_src_lib.c.o): In function `clock_now':
/__w/rizin/rizin/rizin/build/../subprojects/tree-sitter-0.19.4/lib/src/././clock.h:73: undefined reference to `clock_gettime'
/__w/rizin/rizin/rizin/build/../subprojects/tree-sitter-0.19.4/lib/src/././clock.h:73: undefined reference to `clock_gettime'
collect2: error: ld returned 1 exit status
[941/1667] Compiling C object librz/analysis/librz_analysis.so.0.3.0-git.p/data.c.o
[942/1667] Compiling C object librz/asm/librz_asm.so.0.3.0-git.p/arch_hexagon_hexagon_disas.c.o
```

File: librz/socket/run.c [hunk 1]
```diff
@@ -1245,13 +1245,12 @@ RZ_API int rz_run_start(RzRunProfile *p) {
 				}
 			}
 			setsid();
-#if !LIBC_HAVE_FORK
+#if !HAVE_FORK
```
Comment (ret2libc) [inline librz/socket/run.c]: mmm shouldn't this be 
```suggestion
#if HAVE_EXECV
```
?

Comment (XVilka) [reply librz/socket/run.c]: Ｙｅｓ，　ｇｏｏｄ　ｃａｔｃｈ．

**Review by ret2libc — APPROVED** [review]

**Pull request merged**